nvse/nvse/ScriptAnalyzer.cpp

```diff
@@ -8,13 +8,15 @@
 #include "FunctionScripts.h"
 #include "ScriptUtils.h"
 
-std::stack<ScriptParsing::ScriptAnalyzer*> g_analyzerStack;
+std::vector<ScriptParsing::ScriptAnalyzer*> g_analyzerStack;
 
-ScriptParsing::ScriptAnalyzer* GetAnalyzer()
+ScriptParsing::ScriptAnalyzer* GetRootAnalyzer()
 {
-	if (g_analyzerStack.empty())
+	if (g_analyzerStack.empty()) {
 		return nullptr;
-	return g_analyzerStack.top();
+	}
+	
+	return g_analyzerStack[0];

```

Comment:
Is it intended that this function will only ever return the first ScriptAnalyzer to be pushed onto the vector, since the rest of the code is using push_back and pop_back?

Also, what was the point in general to use std::vector over std::stack? Performance?

Reply:
Yes, lambdas and nested lambdas should add their vars to the root scope since that is the script they are in.

Also, cannot index a stack.